deferred_heap.h

```diff
@@ -87,43 +109,38 @@ namespace gcpp {
 			dtors.clear();
 		}
 
-		//	Runn all the destructors for objects in [begin,end)
+		//	Run all the destructors for objects in [begin,end)
 		//
 		bool run(gsl::span<byte> range) {
 			if (range.size() == 0)
 				return false;
 
-			bool ret = false;
-
 			//	for reentrancy safety, we'll take a local copy of destructors to be run
 			//
-			//	first, move any destructors for objects in this range to a local list...
+			//	move any destructors for objects in this range to a local list...
 			//
-			std::vector<destructor> to_destroy;
-			for (auto it = dtors.begin(); it != dtors.end(); /*--*/) {
-				//	<= and -- to avoid dereferencing a past-the-end iterator
-				if (&*range.begin() <= it->p && it->p <= &*--range.end()) {
```

Comment:
`++` and `--` require lvalues when applied to non-class types ([expr.post.incr]/1 "The operand shall be a modifiable lvalue.") . Some of us who are troublemakers like to `&`-qualify the definition of `++` and `--` in our classes so that they require lvalues as well to, uh, "help" people learn the proper semantics ;)

In practice, it's usually only a problem when an iterator could be a pointer. This isn't the case for `gsl::span` right now, but `span` as proposed for standardization in P0122 could be implemented with pointers for iterators (as can `string`, `string_view`, `vector`, and `array`).